src/naive/date.rs

```diff
@@ -2976,6 +2989,12 @@ mod tests {
             NaiveDate::parse_from_str("2019-01-0", "%Y-%W-%w").ok(),
             NaiveDate::from_ymd_opt(2019, 1, 13),
         );
+
+        assert!(NaiveDate::parse_from_str("2014-5-7T12:34:56", "%Y-%m-%dT%H:%M:%S").is_err());
+        assert!(NaiveDate::parse_from_str("2014-5-7+09:30", "%Y-%m-%dT%z").is_err());
+        assert!(
+            NaiveDate::parse_from_str("2014-5-7T12:34:56+09:30", "%Y-%m-%dT%H:%M:%S%z").is_err()
+        );
```

Comment:
IIUC, shouldn't this include `assert!(...is_ok())` for _Only Year_, _Only Year and Month_?
Similar values should be tested in above function `test_date_from_str`.